cnf-certification-test/operator/util.go

```diff
@@ -0,0 +1,80 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/test-network-function/cnf-certification-test/internal/clientsholder"
+	v1 "k8s.io/api/core/v1"
+
+	olmv1Alpha "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Get the list of pods mananged by the operator - retrieve from the CSV and then check the operator group to find the target namespaces to retrieve the managed pods
+func GetNonClusterOperatorCsvPods(csv *olmv1Alpha.ClusterServiceVersion) ([]v1.Pod, error) {
+	csvName := csv.Name
+	annotations := csv.Annotations
+	operatorGroupName := annotations["olm.operatorGroup"]
+
+	// Get the operator group to get the targetNamespace
+	operatorTargetNamespaces, err := getOperatorGroupTargetNamespaceByName(operatorGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("operator Group for found for %s", csvName)
+	}
+
+	var podList []v1.Pod
+
+	for _, targetNamespace := range operatorTargetNamespaces {
+
+		pods, err := getAllPodsManagedByOperatorGroup(targetNamespace, operatorGroupName)
+		if err != nil {
+			continue
+		}
+		podList = append(podList, pods...)
+	}
+
+	return podList, nil
+}
+
+func getOperatorGroupTargetNamespaceByName(operatorGroupName string) ([]string, error) {
+	client := clientsholder.GetClientsHolder()
+
+	operatorGroup, err := client.OlmClient.OperatorsV1().OperatorGroups(operatorGroupName).Get(context.TODO(), operatorGroupName, metav1.GetOptions{})
```

Comment:
OperatorGroups(operatorGroupName) should be namespace (operatorGroupNamespace) ?

Reply:
Yeah, right. I think the code got messed up a bit during refactoring. I moved stuffs from here to there. Otherwise, the get happened based on operator namespace and operator group name. Thanks!